designer/server/src/main/scala/pl/touk/nussknacker/ui/services/BaseHttpService.scala

```diff
@@ -71,6 +72,31 @@ abstract class BaseHttpService(
   protected def businessError[BUSINESS_ERROR](error: BUSINESS_ERROR) = Left(Left(error))
 
   protected def securityError[SE <: SecurityError](error: SE) = Left(Right(error))
+
+  private type PartialEndpoint[INPUT, OUTPUT, BUSINESS_ERROR, -R] =
+    PartialServerEndpoint[_, LoggedUser, INPUT, Either[BUSINESS_ERROR, SecurityError], OUTPUT, R, Future]
+
+  implicit class ServerLogicExtension[INPUT, OUTPUT, BUSINESS_ERROR, -R](
```

Comment:
I would leave it as it is because I believe it's the most common way of handling logic in our services. The shape of the code simplifies structure of returning errors and although the marker trait is not perfect solution, it solves almost all cases of security in this part of the serverLogic.

Reply:
ok, but maybe we should move `CustomAuthorizationError` trait to `BaseHttpService` because it's not related to endpoint definitions but rather to the http service implementation?

Reply:
Done